plugins/pagerduty/src/components/EntityPagerDutyCard/index.test.tsx

```diff
@@ -0,0 +1,386 @@
+            <EntityProvider entity={entityWithServiceId}>
+              <EntityPagerDutyCard />
+            </EntityProvider>
+          </ApiProvider>,
+        ),
+      );
+      await waitFor(() => !queryByTestId('progress'));
+      expect(
+        getByText('Missing or invalid PagerDuty Token'),
+      ).toBeInTheDocument();
+    });
+
+    it('Handles custom NotFoundError', async () => {
+      mockPagerDutyApi.getServiceByPagerDutyEntity = jest
+        .fn()
+        .mockRejectedValueOnce(new NotFoundError());
+
+      const { getByText, queryByTestId } = render(
+        wrapInTestApp(
+          <ApiProvider apis={apis}>
+            <EntityProvider entity={entityWithServiceId}>
+              <EntityPagerDutyCard />
+            </EntityProvider>
+          </ApiProvider>,
+        ),
+      );
+      await waitFor(() => !queryByTestId('progress'));
+      expect(getByText('PagerDuty Service Not Found')).toBeInTheDocument();
+    });
+
+    it('handles general error', async () => {
+      mockPagerDutyApi.getServiceByPagerDutyEntity = jest
+        .fn()
+        .mockRejectedValueOnce(new Error('An error occurred'));
+      const { getByText, queryByTestId } = render(
+        wrapInTestApp(
+          <ApiProvider apis={apis}>
+            <EntityProvider entity={entityWithServiceId}>
+              <EntityPagerDutyCard />
+            </EntityProvider>
+          </ApiProvider>,
+        ),
+      );
+      await waitFor(() => !queryByTestId('progress'));
+
+      expect(
+        getByText(
+          'Error encountered while fetching information. An error occurred',
+        ),
+      ).toBeInTheDocument();
+    });
+
+    it('disables the Create Incident button', async () => {
+      mockPagerDutyApi.getServiceByPagerDutyEntity = jest
+        .fn()
+        .mockImplementationOnce(async () => ({ service }));
+
+      const { queryByTestId, getByTitle } = render(
+        wrapInTestApp(
+          <ApiProvider apis={apis}>
+            <EntityProvider entity={entityWithServiceId}>
+              <EntityPagerDutyCard />
+            </EntityProvider>
+          </ApiProvider>,
+        ),
+      );
+      await waitFor(() => !queryByTestId('progress'));
+      expect(
+        getByTitle('Must provide an integration-key to create incidents')
+          .className,
+      ).toMatch('disabled');
+    });
+  });
+
+  describe('when entity has all annotations', () => {
+    it('queries by integration key', async () => {
+      mockPagerDutyApi.getServiceByPagerDutyEntity = jest
+        .fn()
+        .mockImplementationOnce(async () => ({ service }));
+
+      const { getByText, queryByTestId } = render(
+        wrapInTestApp(
+          <ApiProvider apis={apis}>
+            <EntityProvider entity={entityWithAllAnnotations}>
+              <EntityPagerDutyCard />
+            </EntityProvider>
+          </ApiProvider>,
+        ),
+      );
+      await waitFor(() => !queryByTestId('progress'));
```

Comment:
you may enjoy https://testing-library.com/docs/guide-disappearance/#waiting-for-disappearance